spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -20,110 +20,126 @@ On Ethereum, there are two types of accounts: externally owned accounts, control
 
-`runTx` executes a transaction after the transaction has been successfully authenticated.
+The receiving chain must implement authentication with regard to ensuring that the incoming messages are sent by the owner of the targeted interchain account. With regards to authorization, it is up to each chain-specific implementation to decide if the hosted interchain accounts have the authority to invoke all of the chain's message types or only a subset. This configuration may be set up at module initialization.
 
-```typescript
-type Tx = object
+### Data Structures
 
-interface IBCAccountModule {
-  tryRegisterIBCAccount(data: Uint8Array)
+```typescript
+interface InterchainAccountModule {
+  // Sending/Controlling side
+  tryRegisterInterchainAccount(data: Uint8Array)
   tryRunTx(chainType: Uint8Array, data: any)
-  createAccount(packet: Packet, data: Uint8Array): Address
+  // Recieving side
+  createAccount(): Address
   deserialiseTx(txBytes: Uint8Array): Tx
   authenticateTx(tx: Tx): boolean
   runTx(tx: Tx): Result
 }
 ```
 
-`IBCAccountPacketData` specifies the type of packet. The `IBCAccountPacketData.data` is arbitrary and can be used for different purposes depending on the value of `IBCAccountPacketData.type`.
+**A chain can implement the entire interface, or decide to implement only the sending or receiving parts of the protocol.**
 
-When the value of `type` is `REGISTER`, `IBCAccountPacketData.data` can be used as information by which the account is created on the host chain. When the value of `type` is `RUNTX`, `IBCAccountPacketData.data` contains the tx bytes.
 
-```typescript
-// `Type` enumerator defines the packet type.
-enum Type {
-  REGISTER,
-  RUNTX
-}
+#### Sending Interface
+The `tryRegisterInterchainAccount` method in the `InterchainAccountModule` interface defines the way to request the creation of an interchain account on a remote chain. The remote chain creates an interchain account using its own account creation logic. Due to the limitation of ordered channels, the recommended way to achieve this when calling `tryRegisterInterchainAccount` is to dynamically bind a new port with the port id set as the address of the owner of the account (if the port is not already bound), invoke `OpenChanInit` via an IBC module which will initiate the handshake process and emit an event signaling to a relayer to generate a new channel between both chains. The remote chain can then create the interchain account in the `ChanOpenTry` callback as part of the channel creation handshake process defined in [ics-4](https://github.com/cosmos/ibc/tree/master/spec/core/ics-004-channel-and-packet-semantics). Once the `chanOpenAck` callback fires the handshake-originating (sending) chain can assume the account registration is successful.  
```

Comment:
Is the reason why the protocol is tied to channel handshake is a requirement that interchain account needs to be mapped to a single channel? What I don't get is how the mapping between interchain account on dst chain, controller on source chain and channel is achieved.

Reply:
I'm planning to push some updates to this tomorrow. I don't think the emitting of the event is necessary here or needs to be part of the protocol. 

> Is the reason why the protocol is tied to channel handshake is a requirement that interchain account needs to be mapped to a single channel? 

This was the idea. But perhaps this does not need to be part of the protocol at all. I'm thinking that we begin registering the account on the `onChanOpenInit` callback rather than wait for the event as mentioned here.